modules/waveform-ffmpeg/src/main/java/org/opencastproject/waveform/ffmpeg/WaveformServiceImpl.java

```diff
@@ -402,8 +403,10 @@ private String createWaveformFilter(Track track, int width, int height, String c
       filterBuilder.append(waveformFilterPre);
       filterBuilder.append(",");
     }
-    if (color != null) {
-      waveformColor =  StringUtils.split(color, "|");
+    String[] waveformOperationColors = ArrayUtils.clone(waveformColor);
```

Comment:
There isn't any reason to copy the array and we don't even need to split the configuration (not that my patch did a better job here). Suggestion:

```java
if (StringUtils.isBlank(color) {
    color = StringUtils.join(Arrays.asList(waveformColor), "|");
}
…
filterBuilder.append(color);
```

Reply:
Ah, good catch, definitely don't need the clone.  I was including the clone to make the operation additive (ie, you get the default, and whatever else you want) rather than absolute (you get the default, or what you say you want).  But that does change the API, so I'll revert that.